Monika After Story/game/sprite-decoder.rpy

```diff
@@ -0,0 +1,346 @@
+        fullcode.append(spcode[index])
+
+        sweatdrop = SWEAT_MAP.get("".join(fullcode), None)
+
+        if sweatdrop is None:
+            return False, 0
+
+        #All checks pass, we have sweat
+        export_dict["sweat"] = sweatdrop
+        return True, 1
+
+    def __process_tears(spcode, index, export_dict, *prefixes):
+        """
+        Processes a tear off the given spcode at the given index
+
+        IN:
+            spcode the spcode to check
+            index the next index to check
+            export_dict - dict to add the sprite data to
+            prefixes letters to prefix the code with
+
+        OUT:
+            Tuple of the following format:
+                [0] - True if the tears were valid, False if not
+                [1] - the number of spots to increase the index by
+        """
+        # next letter must be the tear type
+        fullcode = list(prefixes)
+        fullcode.append(spcode[index])
+
+        tears = TEAR_MAP.get("".join(fullcode), None)
+
+        if tears is None:
+            return False, 0
+
+        #Checks passed. Let's add this
+        export_dict["tears"] = tears
+        return True, 1
+
+    PROCESS_MAP = {
+        "b": __process_blush,
+        "s": __process_s,
+        "t": __process_tears,
+    }
+
+    SUB_PROCESS_MAP = {
+        "s": {
+            "d": __process_sweatdrop,
+        },
+    }
+
+    def parse_exp_to_kwargs(exp):
+        """
+        Converts exp codes to kwargs to pass into mas_drawmonika_rk
+
+        IN:
+            exp - spritecode to convert
+
+        OUT:
+            dict representing the exp as kwargs for mas_drawmonika_rk
+
+        ASSUMES:
+            exp is not in the staticsprite format (not exp_static)
+
+        RAISES:
+            - KeyError if pose, eyes, eyebrows, or mouth is invalid
+            - Exception if optional sprite is invalid
+        """
+        full_code = exp
+        kwargs = dict()
+
+        #First parse arms, taking care of leaning as well here
+        arms = ARM_MAP[exp[0]]
+        if arms == "def":
+            kwargs["lean"] = "def"
+            #We'll also manage leaning's standing sprite fully here
+            kwargs["single"] = SINGLE_MAP.get(exp[-1], "3b")
+
+        else:
+            #Leaning doesn't get sides
+            kwargs["left"], kwargs["right"] = SIDES_MAP[exp[0]]
+
+        kwargs["arms"] = arms
+
+        #Remove the first char
+        exp = exp[1:]
+
+        #Now let's see if we have a head match
+        kwargs["head"] = HEAD_MAP.get("".join((exp[0], exp[1], exp[-1])), "")
+
+        #Now the eyes
+        kwargs["eyes"] = EYE_MAP[exp[0]]
+        exp = exp[1:]
+
+        #Now the eyebrows
+        kwargs["eyebrows"] = EYEBROW_MAP[exp[0]]
+        exp = exp[1:]
+
+        #Next, we'll take care of the mouth
+        kwargs["mouth"] = MOUTH_MAP[exp[-1]]
+        exp = exp[:-1]
+
+        #Since nose is static, just keep that here
+        kwargs["nose"] = "def"
+
```

Comment:
sprite.py doesnt change the exp string until right before the loop, so this should do the same to avoid excess string splicing.